From 73c28263a973671de5bce6b9fe1f5a49e981c114 Mon Sep 17 00:00:00 2001
From: Eliad Peller <eliad@wizery.com>
Date: Tue, 30 Jan 2018 15:11:54 +0800
Subject: [PATCH 16/35] driver_nl80211: support passing PSK on connect

commit 730c5a1d098dbdb5edb6c5a7ed5e6ba84503e94c master.

If the driver advertises WPA_DRIVER_FLAGS_4WAY_HANDSHAKE
support, pass the psk on connect.

Signed-off-by: Eliad Peller <eliadx.peller@intel.com>
---
 src/drivers/driver_nl80211.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/src/drivers/driver_nl80211.c b/src/drivers/driver_nl80211.c
index 1210d4356087..93273d26019d 100644
--- a/src/drivers/driver_nl80211.c
+++ b/src/drivers/driver_nl80211.c
@@ -4953,6 +4953,15 @@ static int nl80211_connect_common(struct wpa_driver_nl80211_data *drv,
 			return -1;
 	}
 
+	/* add psk in case of 4way handshake offload */
+	if (params->psk &&
+	    (drv->capa.flags & WPA_DRIVER_FLAGS_4WAY_HANDSHAKE)) {
+		wpa_printf(MSG_DEBUG, "Adding psk (len=32)");
+		wpa_hexdump_key(MSG_DEBUG, "PSK", params->psk, 32);
+		if (nla_put(msg, NL80211_ATTR_PMK, 32, params->psk))
+			return -1;
+	}
+
 	if (nla_put_flag(msg, NL80211_ATTR_CONTROL_PORT))
 		return -1;
 
-- 
2.1.0

